shared/engine/index.js

```diff
@@ -99,13 +100,18 @@ class Engine {
   // We call onDisconnect handlers only if we've actually disconnected (ie connected once)
   _hasConnected: boolean = false
   // So we can dispatch actions
-  _dispatch: Dispatch
+  static _dispatch: Dispatch
   // Temporary helper for incoming call maps
-  _getState: () => TypedState
+  static _getState: () => TypedState
+  static dispatchWaitingAction = (key: string, waiting: boolean) => {
+    Engine._dispatch(createChangeWaiting({key, increment: waiting}))
+  }
 
   constructor(dispatch: Dispatch, getState: () => TypedState) {
-    this._dispatch = dispatch
-    this._getState = getState
+    // setup some static vars
+    Engine._dispatch = dispatch
+    Engine._getState = getState
+    Session._dispatchWaitingAction = Engine.dispatchWaitingAction
```

Comment:
this is mostly so we don't have to deal with session importing engine and dealing with import cycles / flow cycles